backendCore/aws_helpers/dynamo_db/dynamo_db_utils.py

```diff
@@ -0,0 +1,212 @@
+        }
+        if last_evaluated_key is not None:
+            query_params["ExclusiveStartKey"] = last_evaluated_key
+
+        response = table.query(**query_params)
+        response_items.extend(response["Items"])
+        last_evaluated_key = response.get("LastEvaluatedKey")
+        has_more_items = last_evaluated_key is not None
+
+    return response_items
+
+
+def create_dynamo_item(table_name: str, input_item: dict) -> bool:
+    """
+    Creates item in DynamoDB table, replaces item if partition_key already exists
+
+    Args:
+        table_name (str): Name of DynamoDB table
+        input_item (dict): object to insert into table in the form of a dictionary e.g. {'id': 'fesd', 'name': 'bob'}.
+
+    Returns:
+        bool: True if item was created successfully
+
+    Raises:
+        ValueError: If table_name is not a valid table name, or the input dict is missing the partition key or gsi key (if it exists)
+        Exception: If DynamoDB get_item call fails for any reason, e.g., item not found
+    """
+    # Validation steps
+    if table_name not in ALL_DYANMODB_TABLES.keys():
+        raise ValueError("Invalid table name: " + table_name)
+    partition_key = ALL_DYANMODB_TABLES[table_name]["partition_key"]
+    if input_item.get(partition_key) is None:
+        raise ValueError("Item must have the partition key: " + partition_key)
+    gsi_key = ALL_DYANMODB_TABLES[table_name].get("gsi")
+    if gsi_key and input_item.get(gsi_key) is None:
+        raise ValueError("Item must have the gsi key: " + gsi_key)
+
+    # Create item
+    table = dynamodb.Table(table_name)
+    response = table.put_item(Item=input_item)
+    if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+        raise Exception("Failed to delete item")
+    return True
+
+
+def update_dynamo_item(
+    table_name: str, partition_key_value: str or int, update_object: dict
+) -> bool:
+    """
+    Updates item in DynamoDB table, creates if partition_key does not exist
+
+    Args:
+        table_name (str): Name of DynamoDB table
+        partition_key_value (str): Value of partition key to match
+        update_object (dict): dict containing the columns and values to update into table in the form of a dictionary e.g. {'name': 'bob'}, must not contain partition key
+
+    Returns:
+        bool: True if item was updated successfully
+
+    Raises:
+        ValueError: If table_name is not a valid table name
+        Exception: If DynamoDB get_item call fails for any reason, e.g., item not found
+    """
+    # Validation steps
+    if table_name not in ALL_DYANMODB_TABLES.keys():
+        raise ValueError("Invalid table name: " + table_name)
+    partition_key = ALL_DYANMODB_TABLES[table_name]["partition_key"]
+    if partition_key in update_object.keys():
+        raise ValueError("Cannot update partition key: " + partition_key)
+
+    PREFIX = "#dlp__"
+
+    # create update expression
+    update_expression = "SET " + ", ".join(
```

Comment:
do we anticipate any other use cases of this type of update expression. if so, maybe have a utility file to build this type of expression so that we reduce code duplication and increase reusability

Reply:
This is already achieved through this file, which acts as the general utils file for any db operations. Give a schema and it'll automatically convert it to the right SQL query